src/main/java/com/powsybl/forcelayout/ForceLayout.java

```diff
@@ -159,18 +170,58 @@ public void execute() {
         LOGGER.info("Elapsed time: {}", elapsedTime / 1e9);
     }
 
-    private void applyCoulombsLaw() {
+    private Vector coulombsForce(Vector p1, Vector p2, double repulsion) {
+        Vector distance = p1.subtract(p2);
+        Vector direction = distance.normalize();
+        return direction.multiply(repulsion).divide(distance.magnitudeSquare() * 0.5 + 0.1);
+    }
+
+    private void applyCoulombsLawToPoints() {
         for (Point point : points.values()) {
+            Vector p = point.getPosition();
             for (Point otherPoint : points.values()) {
                 if (!point.equals(otherPoint)) {
-                    Vector distance = point.getPosition().subtract(otherPoint.getPosition());
-                    Vector direction = distance.normalize();
+                    point.applyForce(coulombsForce(p, otherPoint.getPosition(), repulsion));
+                }
+            }
+        }
+    }
 
-                    Vector force = direction.multiply(repulsion).divide(distance.magnitudeSquare() * 0.5 + 0.1);
+    private void applyCoulombsLawToSprings() {
+        for (Point point : points.values()) {
+            Vector p = point.getPosition();
+            for (Spring spring : springs) {
+                Point n1 = spring.getNode1();
+                Point n2 = spring.getNode2();
+                if (!n1.equals(point) && !n2.equals(point)) {
+                    Vector q1 = spring.getNode1().getPosition();
+                    Vector q2 = spring.getNode2().getPosition();
+                    Vector center = q1.add(q2.subtract(q1).multiply(0.5));
+                    Vector force = coulombsForce(p, center, repulsion * springRepulsionFactor);
                     point.applyForce(force);
                 }
             }
         }
+        for (Spring spring : springs) {
+            Point n1 = spring.getNode1();
+            Point n2 = spring.getNode2();
+            Vector p1 = spring.getNode1().getPosition();
+            Vector p2 = spring.getNode2().getPosition();
+            Vector center = p1.add(p2.subtract(p1).multiply(0.5));
+            for (Spring otherSpring : springs) {
+                if (!spring.equals(otherSpring)) {
+                    // Compute the repulsion force between centers of the springs
+                    Vector op1 = otherSpring.getNode1().getPosition();
+                    Vector op2 = otherSpring.getNode2().getPosition();
+                    Vector otherCenter = op1.add(op2.subtract(op1).multiply(0.5));
+                    Vector force = coulombsForce(center, otherCenter, repulsion * springRepulsionFactor);
```

Comment:
Similarly here, we could later apply this force only if the springs intersect.

Reply:
To be explored in a separate pull request.